src/codeu/chat/common/ConversationHeader.java

```diff
@@ -44,23 +46,88 @@ public ConversationHeader read(InputStream in) throws IOException {
           Uuid.SERIALIZER.read(in),
           Uuid.SERIALIZER.read(in),
           Time.SERIALIZER.read(in),
-          Serializers.STRING.read(in)
+          Serializers.STRING.read(in),
+          Serializers.INTEGER.read(in)
       );
 
     }
   };
 
+  // permission levels
+  // member == 1
+  // owner == 2
+  // creator == 3
+  
+  public enum permissions {
+    member(1), owner(2), creator(3);
+	private int permissionLevel;
+	
+	private permissions(int permissionLevel) {
+	  this.permissionLevel = permissionLevel;
+	}
+	
+	public int getPermissionLevel() {
+	  return permissionLevel;
+	}
+  }
+ 
   public final Uuid id;
   public final Uuid owner;
   public final Time creation;
   public final String title;
+  public final Integer defaultByte;
+  public final HashMap<Uuid, Integer> userLevels;
```

Comment:
Just to be clear, it's perfectly fine to use "Integer" here (in fact, you must) and still store/retrieve "int" values. Java is a quirky language.